### PR TITLE
Allow bag dumping into mortars and millstones

### PR DESCRIPTION
**Pull request by Azlier**
## About The Pull Request

Allow the dumping of (usually plant) bags into large mortars and millstones. There's no logical reason it shouldn't be possible to do so and it can already be done with biogenerators and reagent grinders. Let's make things a little more consistent and require fewer repetitive clicks to shuffle items around.

## How This Contributes To The Skyrat Roleplay Experience

It's perfectly sensible to be able to dump a bag into a bowl or mill chute for processing. Less pointless clicking leaves more time for actual (role)playing. Carpal tunnel syndrome sucks, too.

## Proof of Testing

<details>
<summary>Screenshots/Videos</summary>
  
Untitled

</details>

## Changelog

:cl:
qol: added dumping bags into large wooden mortars and millstones
/:cl:

### CODE REVIEW

File: modular_skyrat/modules/primitive_cooking_additions/code/big_mortar.dm [hunk 1]
```diff
@@ -59,6 +59,29 @@
...
+			if(length(contents) >= maximum_contained_items)
+				break
+				
+			if(target_item.juice_results || target_item.grind_results)
+				target_item.forceMove(src)
+
+		if (length(contents) >= maximum_contained_items)
+			balloon_alert(user, "mortar filled")
+		else
+			balloon_alert(user, "bag emptied")
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/primitive_cooking_additions/code/big_mortar.dm]: I feel like it might be a little weird if we have this for when you try and dump stuff and nothing comes out, no?

File: modular_skyrat/modules/primitive_cooking_additions/code/big_mortar.dm [hunk 2]
```diff
@@ -59,6 +59,29 @@
 	return SECONDARY_ATTACK_CANCEL_ATTACK_CHAIN
 
 /obj/structure/large_mortar/attackby(obj/item/attacking_item, mob/living/carbon/human/user)
+	if(istype(attacking_item, /obj/item/storage/bag))
+		if(length(contents) >= maximum_contained_items)
+			balloon_alert(user, "already full")
+			return TRUE
+		
+		if(!attacking_item.contents.len)
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/primitive_cooking_additions/code/big_mortar.dm]:
```suggestion
		if(!length(attacking_item.contents))
```

File: modular_skyrat/modules/primitive_cooking_additions/code/big_mortar.dm [hunk 3]
```diff
@@ -59,6 +59,29 @@
...
+			
+		for(var/obj/item/target_item in attacking_item.contents)
+			if(length(contents) >= maximum_contained_items)
+				break
+				
+			if(target_item.juice_results || target_item.grind_results)
+				target_item.forceMove(src)
+
+		if (length(contents) >= maximum_contained_items)
+			balloon_alert(user, "mortar filled")
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/primitive_cooking_additions/code/big_mortar.dm]:
```suggestion
			balloon_alert(user, "[src] filled")
```

File: modular_skyrat/modules/primitive_cooking_additions/code/big_mortar.dm [hunk 4]
```diff
@@ -59,6 +59,29 @@
 	return SECONDARY_ATTACK_CANCEL_ATTACK_CHAIN
 
 /obj/structure/large_mortar/attackby(obj/item/attacking_item, mob/living/carbon/human/user)
+	if(istype(attacking_item, /obj/item/storage/bag))
+		if(length(contents) >= maximum_contained_items)
+			balloon_alert(user, "already full")
+			return TRUE
+		
+		if(!attacking_item.contents.len)
+			balloon_alert(user, "bag empty")
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/primitive_cooking_additions/code/big_mortar.dm]:
```suggestion
			balloon_alert(user, "[attacking_item] empty")
```

File: modular_skyrat/modules/primitive_cooking_additions/code/millstone.dm [hunk 5]
```diff
@@ -66,6 +66,28 @@
 	return SECONDARY_ATTACK_CANCEL_ATTACK_CHAIN
 
 /obj/structure/millstone/attackby(obj/item/attacking_item, mob/user)
+	if(istype(attacking_item, /obj/item/storage/bag))
+		if(length(contents) >= maximum_contained_items)
+			balloon_alert(user, "already full")
+			return TRUE
+		
+		if(!attacking_item.contents.len)
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/primitive_cooking_additions/code/millstone.dm]:
```suggestion
		if(!length(attacking_item.contents))
```

File: modular_skyrat/modules/primitive_cooking_additions/code/millstone.dm [hunk 6]
```diff
@@ -66,6 +66,28 @@
...
+			balloon_alert(user, "bag empty")
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/primitive_cooking_additions/code/millstone.dm]:
```suggestion
			balloon_alert(user, "[attacking_item] empty")
```

File: modular_skyrat/modules/primitive_cooking_additions/code/millstone.dm [hunk 7]
```diff
@@ -66,6 +66,28 @@
...
+			return TRUE
+			
+		for(var/obj/item/food/grown/target_item in attacking_item.contents)
+			if(length(contents) >= maximum_contained_items)
+				break
+				
+			target_item.forceMove(src)
+		
+		if (length(contents) >= maximum_contained_items)
+			balloon_alert(user, "millstone filled")
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/primitive_cooking_additions/code/millstone.dm]:
```suggestion
			balloon_alert(user, "[src] filled")
```

File: modular_skyrat/modules/primitive_cooking_additions/code/millstone.dm [hunk 8]
```diff
@@ -66,6 +66,28 @@
...
+		else
+			balloon_alert(user, "bag emptied")
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/primitive_cooking_additions/code/millstone.dm]:
```suggestion
			balloon_alert(user, "[attacking_item] emptied")
```

Comment (GoldenAlpharex) [inline modular_skyrat/modules/primitive_cooking_additions/code/millstone.dm]: Same comment as for the other one, though. What if you try to put stuff in it and nothing comes out, or there's still stuff left in your bag?

I feel like we could probably use the code for dumping things into a container using click-drag instead, which already handles almost all of that (minus I guess a check we'd have to add to it for this), no?

**Review by GoldenAlpharex — DISMISSED** [review]

File: modular_skyrat/modules/primitive_cooking_additions/code/big_mortar.dm (hunk 1 above)
Comment (Azlier) [reply modular_skyrat/modules/primitive_cooking_additions/code/big_mortar.dm]: That's exactly the behavior shown when dumping a plant bag into a biogenerator, yes. It's a little weird but I wanted to keep it consistent between functionally similar structures. I also avoided using [attacking_item] and [src] for reason of keeping balloon text short and sweet because 'large mortar filled' and 'portable seed extractor emptied' were just a little too wordy for my tastes, but that's fine.

File: modular_skyrat/modules/primitive_cooking_additions/code/millstone.dm [hunk 9]
```diff
@@ -66,6 +66,28 @@
...
+			return TRUE
+			
+		for(var/obj/item/food/grown/target_item in attacking_item.contents)
+			if(length(contents) >= maximum_contained_items)
+				break
+				
+			target_item.forceMove(src)
+		
+		if (length(contents) >= maximum_contained_items)
+			balloon_alert(user, "[src] filled")
```
Comment (lessthnthree) [inline modular_skyrat/modules/primitive_cooking_additions/code/millstone.dm]:
```suggestion
			balloon_alert(user, "filled!")
```

Comment (lessthnthree) [reply modular_skyrat/modules/primitive_cooking_additions/code/millstone.dm]: We don't use source object names in balloon alerts

File: modular_skyrat/modules/primitive_cooking_additions/code/millstone.dm [hunk 10]
```diff
@@ -66,6 +66,28 @@
 	return SECONDARY_ATTACK_CANCEL_ATTACK_CHAIN
 
 /obj/structure/millstone/attackby(obj/item/attacking_item, mob/user)
+	if(istype(attacking_item, /obj/item/storage/bag))
+		if(length(contents) >= maximum_contained_items)
+			balloon_alert(user, "already full")
+			return TRUE
+		
+		if(!length(attacking_item.contents))
+			balloon_alert(user, "[attacking_item] empty")
```
Comment (lessthnthree) [inline modular_skyrat/modules/primitive_cooking_additions/code/millstone.dm]:
```suggestion
			balloon_alert(user, "nothing to transfer!")
```

File: modular_skyrat/modules/primitive_cooking_additions/code/millstone.dm [hunk 11]
```diff
@@ -66,6 +66,28 @@
...
+			return TRUE
+
```
Comment (lessthnthree) [inline modular_skyrat/modules/primitive_cooking_additions/code/millstone.dm]:
```suggestion
			balloon_alert(user, "transferring...")
```

File: modular_skyrat/modules/primitive_cooking_additions/code/millstone.dm [hunk 12]
```diff
@@ -66,6 +66,28 @@
...
+		for(var/obj/item/food/grown/target_item in attacking_item.contents)
+			if(length(contents) >= maximum_contained_items)
+				break
+				
+			target_item.forceMove(src)
+		
+		if (length(contents) >= maximum_contained_items)
+			balloon_alert(user, "[src] filled")
+		else
+			balloon_alert(user, "[attacking_item] emptied")
```
Comment (lessthnthree) [inline modular_skyrat/modules/primitive_cooking_additions/code/millstone.dm]:
```suggestion
			balloon_alert(user, "transferred")
```

Comment (lessthnthree) [reply modular_skyrat/modules/primitive_cooking_additions/code/millstone.dm]: this will yield "The chemistry bag emptied" and why we don't use it. don't assume the bag is empty, it also accounts for if you try to dump in things that it can't mill.

File: modular_skyrat/modules/primitive_cooking_additions/code/millstone.dm (hunk 10 above)
Comment (lessthnthree) [reply modular_skyrat/modules/primitive_cooking_additions/code/millstone.dm]: don't use [src] or other name vars in balloon alerts unless there's exactly one context. this yields "The chemistry bag empty"

File: modular_skyrat/modules/primitive_cooking_additions/code/millstone.dm (hunk 11 above)
Comment (lessthnthree) [reply modular_skyrat/modules/primitive_cooking_additions/code/millstone.dm]: there may be items in the bag that can't be transferred, for example a bag filled with glass beakers will attempt to transfer.

File: modular_skyrat/modules/primitive_cooking_additions/code/big_mortar.dm [hunk 13]
```diff
@@ -59,6 +59,29 @@
 	return SECONDARY_ATTACK_CANCEL_ATTACK_CHAIN
 
 /obj/structure/large_mortar/attackby(obj/item/attacking_item, mob/living/carbon/human/user)
+	if(istype(attacking_item, /obj/item/storage/bag))
+		if(length(contents) >= maximum_contained_items)
+			balloon_alert(user, "already full")
```
Comment (lessthnthree) [inline modular_skyrat/modules/primitive_cooking_additions/code/big_mortar.dm:64]: see comment on millstone

File: modular_skyrat/modules/primitive_cooking_additions/code/big_mortar.dm [hunk 14]
```diff
@@ -59,6 +59,29 @@
...
+			return TRUE
+		
+		if(!length(attacking_item.contents))
+			balloon_alert(user, "[attacking_item] empty")
```
Comment (lessthnthree) [inline modular_skyrat/modules/primitive_cooking_additions/code/big_mortar.dm]: see comment on millstone

File: modular_skyrat/modules/primitive_cooking_additions/code/big_mortar.dm [hunk 15]
```diff
@@ -59,6 +59,29 @@
...
+			
+		for(var/obj/item/target_item in attacking_item.contents)
+			if(length(contents) >= maximum_contained_items)
+				break
+				
+			if(target_item.juice_results || target_item.grind_results)
+				target_item.forceMove(src)
+
+		if (length(contents) >= maximum_contained_items)
+			balloon_alert(user, "[src] filled")
```
Comment (lessthnthree) [inline modular_skyrat/modules/primitive_cooking_additions/code/big_mortar.dm]: see comment on millstone

File: modular_skyrat/modules/primitive_cooking_additions/code/big_mortar.dm [hunk 16]
```diff
@@ -59,6 +59,29 @@
...
+		else
+			balloon_alert(user, "bag emptied")
```
Comment (lessthnthree) [inline modular_skyrat/modules/primitive_cooking_additions/code/big_mortar.dm]: see comment on millstone

**Review by lessthnthree — CHANGES_REQUESTED** [review]

**Review by lessthnthree — APPROVED** [review]

**Comment by softcerv** [comment]
Fuck
